src/wallet/walletdb.cpp

```diff
@@ -1048,9 +1052,11 @@ std::unique_ptr<WalletDatabase> MakeDatabase(const fs::path& path, const Databas
     // Format is not set when a db doesn't already exist, so use the format specified by the options if it is set.
     if (!format && options.require_format) format = options.require_format;
 
+#ifdef USE_SQLITE
     if (format && format == DatabaseFormat::SQLITE) {
         return MakeSQLiteDatabase(path, options, status, error);
     }
+#endif
 
     return MakeBerkeleyDatabase(path, options, status, error);
```

Comment:
nit, assert  `format == DatabseFormat::BERKELEY`.

Reply:
It can also be unset. Used `assert(format != DatabaseFormat::SQLITE);`  in an `#else`